bitcoin/src/address.rs

```diff
@@ -505,13 +526,12 @@ impl<V: NetworkValidation> Address<V> {
             Network::Bitcoin => SCRIPT_ADDRESS_PREFIX_MAIN,
             Network::Testnet | Network::Signet | Network::Regtest => SCRIPT_ADDRESS_PREFIX_TEST,
         };
-        let bech32_hrp = match self.network() {
-            Network::Bitcoin => "bc",
-            Network::Testnet | Network::Signet => "tb",
-            Network::Regtest => "bcrt",
+        let hrp = match self.network() {
+            Network::Bitcoin => Hrp::parse("bc").expect("bc is valid"),
+            Network::Testnet | Network::Signet => Hrp::parse("tb").expect("tb is valid"),
+            Network::Regtest => Hrp::parse("bcrt").expect("bcrt is valid"),
```

Comment:
In 4321612b0015f9a09ca3f95f2256cdb7ae2d5876:

Let's expose all these as constants like `Hrp::BC` `Hrp::BRCT` `Hrp::TB`. It's a little ugly in that it exposes segwit stuff in "generic" code but I think it's fine because rust-bitcoin *is* the primary usecase for the library, and it doesn't hurt anybody else to have a bit of extra API.

Reply:
Yep, good idea.

Reply:
I used `hrp::BC` etc and also added functions `Hrp::is_valid_on_mainnet()` and others for testnet, signet, regtest.